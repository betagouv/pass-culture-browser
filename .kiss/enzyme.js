import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
