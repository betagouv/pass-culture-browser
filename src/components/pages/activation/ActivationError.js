/* eslint
  react/jsx-one-expression-per-line: 0 */
import React from 'react'

import MailToLink from '../../layout/MailToLink'
import { SUPPORT_EMAIL } from '../../../utils/config'

const ActivationError = () => (
  <div className="padded-2x flex-1 flex-rows flex-center">
    <h2 className="fs20">Il semblerait que le lien cliqu√© soit incorrect.</h2>
    <p className="fs20">
      <MailToLink
        obfuscate
        email={SUPPORT_EMAIL}
        id="activation-error-contact-us"
        className="no-underline is-block is-white-text py12"
      >
        <span className="is-underline">Contactez-nous</span>
      </MailToLink>
    </p>
  </div>
)
export default ActivationError
