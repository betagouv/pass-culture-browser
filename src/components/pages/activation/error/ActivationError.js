/* eslint
  react/jsx-one-expression-per-line: 0 */
import React from 'react'

import MailToLink from '../../../layout/MailToLink'
import { SUPPORT_EMAIL, SUPPORT_EMAIL_SUBJECT } from '../../../../utils/config'

const ActivationError = () => {
  const emailHeaders = {
    subject: decodeURI(SUPPORT_EMAIL_SUBJECT),
  }
  return (
    <main
      className="pc-main padded-2x flex-rows flex-center"
      id="activation-error-page"
      role="main"
    >
      <div className="flex-center flex-row">
        <p className="fs20">{'Il semblerait que le lien cliqu√© soit incorrect.'}</p>
      </div>

      <div className="flex-center flex-row padded">
        <MailToLink
          className="no-background border-all rd4 py12 px18 is-inline-block is-white-text text-center fs16"
          email={SUPPORT_EMAIL}
          headers={emailHeaders}
          id="activation-error-contact-us"
          obfuscate
        >
          <span>{'Contactez-nous'}</span>
        </MailToLink>
      </div>
    </main>
  )
}

export default ActivationError
