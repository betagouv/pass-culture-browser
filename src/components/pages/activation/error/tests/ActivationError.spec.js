// jest --env=jsdom ./src/components/pages/activation/error/tests/index --watch
import React from 'react'
import { render } from 'enzyme'
import ActivationError from '../ActivationError'

describe('src | components | pages | activation | InvalidLink', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = render(<ActivationError />)

    // then
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should contains a link with an href attributes equal to mailto:obfuscated', () => {
    // when
    const wrapper = render(<ActivationError />)
    const element = wrapper.find('#activation-error-contact-us')

    // then
    expect(element.prop('href')).toEqual('mailto:obfuscated')
  })
})
