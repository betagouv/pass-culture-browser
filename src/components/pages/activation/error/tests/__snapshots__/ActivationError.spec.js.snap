// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | activation | ActivationError should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivationError />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex-center flex-row"
        >
          <p
            className="fs20"
          >
            Il semblerait que le lien cliqué soit incorrect.
          </p>
        </div>,
        <div
          className="flex-center flex-row padded"
        >
          <MailToLink
            className="no-background border-all rd4 py12 px18 is-inline-block is-white-text text-center fs16"
            email="support@passculture.app"
            headers={
              Object {
                "subject": "Votre question depuis l'application pass Culture",
              }
            }
            id="activation-error-contact-us"
            obfuscate={true}
          >
            <span>
              Contactez-nous
            </span>
          </MailToLink>
        </div>,
      ],
      "className": "pc-main padded-2x flex-rows flex-center",
      "id": "activation-error-page",
      "role": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="fs20"
          >
            Il semblerait que le lien cliqué soit incorrect.
          </p>,
          "className": "flex-center flex-row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Il semblerait que le lien cliqué soit incorrect.",
            "className": "fs20",
          },
          "ref": null,
          "rendered": "Il semblerait que le lien cliqué soit incorrect.",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <MailToLink
            className="no-background border-all rd4 py12 px18 is-inline-block is-white-text text-center fs16"
            email="support@passculture.app"
            headers={
              Object {
                "subject": "Votre question depuis l'application pass Culture",
              }
            }
            id="activation-error-contact-us"
            obfuscate={true}
          >
            <span>
              Contactez-nous
            </span>
          </MailToLink>,
          "className": "flex-center flex-row padded",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span>
              Contactez-nous
            </span>,
            "className": "no-background border-all rd4 py12 px18 is-inline-block is-white-text text-center fs16",
            "email": "support@passculture.app",
            "headers": Object {
              "subject": "Votre question depuis l'application pass Culture",
            },
            "id": "activation-error-contact-us",
            "obfuscate": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Contactez-nous",
            },
            "ref": null,
            "rendered": "Contactez-nous",
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex-center flex-row"
          >
            <p
              className="fs20"
            >
              Il semblerait que le lien cliqué soit incorrect.
            </p>
          </div>,
          <div
            className="flex-center flex-row padded"
          >
            <MailToLink
              className="no-background border-all rd4 py12 px18 is-inline-block is-white-text text-center fs16"
              email="support@passculture.app"
              headers={
                Object {
                  "subject": "Votre question depuis l'application pass Culture",
                }
              }
              id="activation-error-contact-us"
              obfuscate={true}
            >
              <span>
                Contactez-nous
              </span>
            </MailToLink>
          </div>,
        ],
        "className": "pc-main padded-2x flex-rows flex-center",
        "id": "activation-error-page",
        "role": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="fs20"
            >
              Il semblerait que le lien cliqué soit incorrect.
            </p>,
            "className": "flex-center flex-row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Il semblerait que le lien cliqué soit incorrect.",
              "className": "fs20",
            },
            "ref": null,
            "rendered": "Il semblerait que le lien cliqué soit incorrect.",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <MailToLink
              className="no-background border-all rd4 py12 px18 is-inline-block is-white-text text-center fs16"
              email="support@passculture.app"
              headers={
                Object {
                  "subject": "Votre question depuis l'application pass Culture",
                }
              }
              id="activation-error-contact-us"
              obfuscate={true}
            >
              <span>
                Contactez-nous
              </span>
            </MailToLink>,
            "className": "flex-center flex-row padded",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                Contactez-nous
              </span>,
              "className": "no-background border-all rd4 py12 px18 is-inline-block is-white-text text-center fs16",
              "email": "support@passculture.app",
              "headers": Object {
                "subject": "Votre question depuis l'application pass Culture",
              },
              "id": "activation-error-contact-us",
              "obfuscate": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Contactez-nous",
              },
              "ref": null,
              "rendered": "Contactez-nous",
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
